### PR TITLE
chore: release 0.25.0

### PR DESCRIPTION
**Pull request by guilhem-barthes**
Signed-off-by: Guilhem Barthes <guilhem.barthes@owkin.com>

### CODE REVIEW

File: docs/source/additional/releases.yaml [hunk 1]
```diff
@@ -9,6 +9,44 @@ components: # for table headers
   - substra-tests
 
 releases:
+  - version: 0.25.0
+    components:
+      substrafl:
+        version: 0.43.0
```
Comment (RomainGoussault) [inline docs/source/additional/releases.yaml:15]:
```suggestion
        version: 0.34.0
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -21,18 +21,20 @@ jobs:
...
           repository: substra/substra
           path: substra
+          ref: release/0.42.0
 
       - name: Clone substrafl
         uses: actions/checkout@v2
         with:
           repository: substra/substrafl
-          ref: main
+          ref: release/0.34.0
```
Comment (guilhem-barthes) [inline .github/workflows/build.yml:37]: Remove before release

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -21,18 +21,20 @@ jobs:
...
           repository: substra/substra-tools
           path: substra-tools
+          ref: 0.20.0
 
       - name: Clone substra
         uses: actions/checkout@v2
         with:
           repository: substra/substra
           path: substra
+          ref: release/0.42.0
```
Comment (guilhem-barthes) [inline .github/workflows/build.yml:31]: Remove before release

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -21,18 +21,20 @@ jobs:
         with:
           repository: substra/substra-tools
           path: substra-tools
+          ref: 0.20.0
```
Comment (guilhem-barthes) [inline .github/workflows/build.yml:24]: Remove before release

File: docs/source/conf.py [hunk 5]
```diff
@@ -27,14 +27,14 @@
...
-    SUBSTRAFL_VERSION = "0.33.0"
+    SUBSTRAFL_VERSION = "0.34.0"
 
 else:
-    SUBSTRA_VERSION = "main"
-    TOOLS_VERSION = "main"
-    SUBSTRAFL_VERSION = "main"
+    SUBSTRA_VERSION = "release/0.42.0"
+    TOOLS_VERSION = "0.20.0"
+    SUBSTRAFL_VERSION = "release/0.34.0"
```
Comment (guilhem-barthes) [inline docs/source/conf.py:37]: Change back to `main` before release

Comment (ThibaultFy) [reply docs/source/conf.py:37]: You need to put 0.42.0rc1 and 0.34.0rc1 to test the release. SubstraFL and Substra are officially release in the last step :)

**Review by ThibaultFy — APPROVED** [review]

**Pull request merged**